leo_fw/src/leo_fw/test_hw.py

```diff
@@ -0,0 +1,410 @@
+from typing import Optional
+from enum import Enum
+
+import time
+
+import rospy
+import rosgraph
+import rosnode
+import rospkg
+
+import yaml
+
+from .utils import *
+from .board import BoardType, determine_board, check_firmware_version
+from leo_msgs.msg import Imu, WheelStates
+from geometry_msgs.msg import Twist
+from std_msgs.msg import Float32
+
+rospack = rospkg.RosPack()
+
+###YAML PARSER
+
+
+def parse_yaml(file_path):
+    with open(file_path, "r") as stream:
+        try:
+            return yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    return {}
+
+
+path = rospack.get_path("leo_fw")+"/test_data/"
+
+imuData = Imu()
+wheelData = WheelStates()
+batteryData = Float32()
+
+isNewImuData = 0
+isNewWheelData = 0
+isNewBatteryData = 0
+
+isWheelLoaded = 1
```

Comment:
Global variables are bad practice, especially in modules. If you want to a shared state for functions, maybe try wrapping them in a class (e.g. `HardwareTester`).